Documentation/BabylonCommands.md

```diff
@@ -0,0 +1,110 @@
+# Babylon's Stevenson Commands
+
+This Vapor project consists actually of two projects:
+
+ * The Stevenson framework, aimed to be reused so you can create your own bots interacting with Slack, GitHub, CircleCI and JIRA yourselves
+ * The Stevenson Babylon app, which is the instance of the Bot we use at Babylon, based on the Stevenson framework, and which provides us with the Slack commands we need
+
+# ⚙️ Environment Variables used by our app
+
+The bot instance of Stevenson we use at Babylon uses environment variables to instantiate and configure the various services (see [README.md](../README.md#supported-services)).
+
+See `app.swift` for declaration of those environment variables used in the app, and `configure.swift` for where they are used to instantiate the services.
+
+ENV Var | Description
+--------|--------------
+`SLACK_TOKEN` | Your access token for the Slack API of your team's Slack
+`GITHUB_USERNAME`<br/>`GITHUB_TOKEN`| Login and access token of a user having read access to those repositories
+`CIRCLECI_TOKEN`| Your access token for the CircleCI API
+`JIRA_BASEURL`| Host name of your Jira instance (e.g. `https://myorg.atlassian.net:443`)
+`JIRA_USERNAME`<br/>`JIRA_TOKEN`| Login and access token for the API of your Jira instance
+
+# 🕹 Commands implemented in the Babylon Bot
+
+The Stevenson instance used at Babylon, whose code can be found in the `App/` directory, declares the features described below, which mainly consist of 3 main core features:
+
+* Invoking a `fastlane` command on CI
+* Responding to a GitHub comment to trigger a CI pipeline
+* [Generating the CRP ticket](CRP.md)
+
+## `/stevenson`
+
+This is the main Slack command, used to namespace other commands. It's implemented in `MainCommand.swift`
+
+This command expects one of the following sub-commands, which are described in more details below:
+
+* `/stevenson fastlane …`
+* `/stevenson appcenter …`
+* `/stevenson testflight …`
+* `/stevenson crp …`
+
+## `/stevenson fastlane`
+
+This and the 2 following commands are implemented in `SlackCommand+Fastlane.swift`
+> Also still accessible via `/fastlane` directly, though deprecated in favor of being invoked as a `/stevenson` subcommand instead.
+
+This command allows you to trigger a CircleCI job which will run the provided lane.
+
+```
+/stevenson fastlane <lane> [branch:<branch>] [target:<target>] [<other_options>]
+```
+
+* If `branch:` is not provided, defaults to running the lane on `develop`
+* `target:` is only used by some lanes in Fastlane, like `appcenter` or `testflight`. In most cases it's not needed
+* Any potential `<other_options>` depend on each lane. Check our Fastlane to see which are available for each. Examples include `version:` for `appcenter` and `testflight` lanes
```

Comment:
full examples would help understanding the command. Especially for people unfamiliar with this kind of flow

Reply:
I didn't want those to be too much detailed in here – as I was trying to focus on technical details and implementation, which any _usage documentation_ of babylon-specific commands should be documented in the playbook, not here.

tbh I was even hesitant to keep that `BabylonCommands.md` document here in the repo – its goal was just to describe in a technical POV examples of commands that we could (and did) implement with the `Stevenson.framework`, for anyone else in the OSS community that would be interested to use those as examples/inspiration to implement their own bot using Stevenson-the-framework

I could totally add examples in here but I feel it would make more sense to keep detailed usage documentation in our playbook. Maybe we could even try to remove some paragraphs in that `BabylonCommands.md` doc to really only focus on the implementation details rather than usage description, but it's hard to talk about implementation without explaining even a little the expected thing we are trying to parse 😉

Reply:
I thought this work was strictly about the process ticket CNSMR-3272. That doc should focus more on the usage rather then implementation. It would be great if you can keep this document that you have just created and have a shorter version focused on the usage. What do you think?

Reply:
Yeah the separation as to where document things is not always clear indeed.

We agreed that the documentation should ideally be split in:
 - General Technical documentation about the framework itself (intended for users of the OSS framework) ➡️ in the repo itself
 - Anything specific to Babylon (especially process and specific commands) ➡️ in the playbook

Maybe I can just move this `BabylonCommands.md` doc in the playbook after all then since it's about the usage and specific to commands implemented for Babylon (and not doc for the bot framework intended for OSS users). But **we already have documentation in the playbook describing those commands [here](https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/SlackCIIntegration.md) though** (as well as in the playbook doc about release process) so it would not make sense to duplicate the documentation.

Maybe I will just delete that `BabylonCommands.md` doc entirely then since it doesn't really talk about implementation details after all (and usage doc already exists in playbook), wdyt?

Reply:
@viorel15 as discussed in Slack, cleaned up the `BabylonCommands.md` to only focus on implementation details, and links to usage in playbook otherwise.

Still kept the document since we needed a place especially to document the environment variables used by our bot instance (being technical documentation for the bot, I believe that belonged more in this repo than the playbook), but otherwise simplified the doc to only point to relevant places.

Reply:
@AliSoftware if that's the only purpose of this doc that's left (to me it feels it does not add much more clarity about implementation details that we already have in other docs or in code except describing the structure of the project) wouldn't be better to document environment variables in code instead of a separate document and then remove this doc? We can have a small section in the main README dedicated to the structure of the project and references that we have in a playbook.